TF-4186 Remember choose organisation server

dab246 opened the pull request:
## Issue

#4186 

## Resolved


https://github.com/user-attachments/assets/ee02fb24-827b-480c-9660-90e94839dad2



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Save, retrieve and remove company server login info on mobile; add login argument support to pass saved info.
  * Auto-fill previous company server email in the DNS login flow.

* **Improvements**
  * Recent-username suggestions now filter by typed input.
  * Mobile login navigation and flow refined for clearer DNS/login transitions and session synchronization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dab246 committed:
TF-4186 Add datasource/repository for save and get company server login info

hoangdat commented:
- [ ] integration test to make sure user should not go to `Welcome screen` if he logged in in organization before

hoangdat reviewed (commented):
- [x] should handle for the case: user should automatically jump to login when open app from terminated

hoangdat commented:
- [ ] shoud also auto fill email when user back to welcome screen -> go to `user company server` again

dab246 commented on test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:
```diff
@@ -257,6 +259,7 @@ void main() {
   final deleteCredentialInteractor = MockDeleteCredentialInteractor();
   final logoutOidcInteractor = MockLogoutOidcInteractor();
   final deleteAuthorityOidcInteractor = MockDeleteAuthorityOidcInteractor();
+  final detCompanyServerLoginInfoInteractor = MockGetCompanyServerLoginInfoInteractor();
```
Fixed

dab246 commented:
> * [x]  should handle for the case: user should automatically jump to login when open app from terminated

Updated

[Screen_recording_20251204_144743.webm](https://github.com/user-attachments/assets/8958d17c-a890-446f-ae4a-ca6677dc0615)


> * [x]  shoud also auto fill email when user back to welcome screen -> go to `user company server` again

Updated


[Screen_recording_20251204_154058.webm](https://github.com/user-attachments/assets/a579c92a-e2d7-4b6b-9fca-df0e3bfb9f3b)

chibenwa commented:
<img width="1335" height="374" alt="Screenshot from 2025-12-05 17-19-32" src="https://github.com/user-attachments/assets/96a4c948-fff0-4f0e-a314-9591504ceddc" />

Having so much changes for just a simple change seems absurd to me and points to a critical lack of abstractions in the current code base.

While this is by no means to *that* PR to address this I would like the team to raise awareness on those concerns.

dab246 committed:
TF-4186 Implement remember choose organisation server

dab246 committed:
TF-4186 Use `super.props` to include parent's properties in the props override.

dab246 committed:
fixup! TF-4186 Use `super.props` to include parent's properties in the props override.

hoangdat commented on lib/features/login/presentation/login_controller.dart:
```diff
@@ -555,6 +576,10 @@ class LoginController extends ReloadableController {
   void _handleCommonOIDCFailure() {
+    if (PlatformInfo.isMobile) {
+      removeCompanyServerLoginInfo();
+    }
```
why we need to remove company server here?

dab246 replied:
we should delete when user enters server url, not through dns
